File: hooks/post-auto-setup.py
#!/usr/bin/env python3
"""PostToolUse hook (Bash): run setup.py after git pull/commit in ~/.claude."""

import json
import re
import subprocess
import sys


def main() -> None:
    data = json.load(sys.stdin)
    cwd = data.get("cwd", "")
    command = data.get("tool_input", {}).get("command", "")

    if not re.search(r"\bgit\s+(pull|commit)\b", command):
        return

    normalized = cwd.replace("\\", "/").rstrip("/")
    if not normalized.endswith("/.claude"):
        return

    result = subprocess.run(
        ["uv", "run", "python", "setup.py"],
        capture_output=True,
        text=True,
        cwd=cwd,
        timeout=30,
    )

    output = result.stdout.strip() or result.stderr.strip()
    if output:
        json.dump(
            {
                "hookSpecificOutput": {
                    "hookEventName": "PostToolUse",
                    "additionalContext": f"[auto] setup.py executed:\n{output}",
                }
            },
            sys.stdout,
        )


if __name__ == "__main__":
    main()

File: hooks-rs/src/hooks/post_auto_setup.rs
use regex::Regex;
use serde_json::json;
use std::io::{self, Write};
use std::time::Duration;

use crate::input::HookInput;
use crate::process::output_with_timeout;

pub fn run(input: &HookInput) {
    let cwd = &input.cwd;
    let command = &input.tool_input.command;
    if command.is_empty() || cwd.is_empty() {
        return;
    }

    if !Regex::new(r"\bgit\s+(pull|commit)\b")
        .unwrap()
        .is_match(command)
    {
        return;
    }
    let normalized = cwd.replace('\\', "/").trim_end_matches('/').to_string();
    if !normalized.ends_with("/.claude") {
        return;
    }

    let current_exe = match std::env::current_exe() {
        Ok(path) => path,
        Err(_) => return,
    };

    let mut command = std::process::Command::new(current_exe);
    command.arg("setup").current_dir(cwd);

    let output = match output_with_timeout(&mut command, Duration::from_secs(30)) {
        Ok(o) => o,
        Err(_) => return,
    };

    let stdout = String::from_utf8_lossy(&output.stdout).trim().to_string();
    let stderr = String::from_utf8_lossy(&output.stderr).trim().to_string();
    let text = if !stdout.is_empty() { stdout } else { stderr };
    if !text.is_empty() {
        let msg = json!({"hookSpecificOutput": {"hookEventName": "PostToolUse", "additionalContext": format!("[auto] claude-hooks setup executed:\n{text}")}});
        let _ = writeln!(io::stdout(), "{msg}");
    }
}
